packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2464,6 +2489,397 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+    properties.add(StringProperty('label', label));
+  }
+}
+
+class _MenuAcceleratorLabelState extends State<MenuAcceleratorLabel> {
+  late String _displayLabel;
+  int _acceleratorIndex = -1;
+  MenuAcceleratorCallbackBinding? _binding;
+  _MenuAnchorState? _anchor;
+  ShortcutRegistry? _shortcutRegistry;
+  ShortcutRegistryEntry? _shortcutRegistryEntry;
+  bool _showAccelerators = false;
+
+  @override
+  void initState() {
+    super.initState();
+    if (_platformSupportsAccelerators()) {
+      _showAccelerators = _altIsPressed();
+      HardwareKeyboard.instance.addHandler(_handleKeyEvent);
+    }
+    _updateDisplayLabel();
+  }
+
+  @override
+  void dispose() {
+    assert(_platformSupportsAccelerators() || _shortcutRegistryEntry == null);
+    _displayLabel = '';
+    if (_platformSupportsAccelerators()) {
+      _shortcutRegistryEntry?.dispose();
+      _shortcutRegistryEntry = null;
+      _shortcutRegistry = null;
+      _anchor = null;
+      HardwareKeyboard.instance.removeHandler(_handleKeyEvent);
+    }
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_platformSupportsAccelerators()) {
+      return;
+    }
+    _binding = MenuAcceleratorCallbackBinding.maybeOf(context);
+    _anchor = _MenuAnchorState._maybeOf(context);
+    _shortcutRegistry = ShortcutRegistry.maybeOf(context);
+    _updateAcceleratorShortcut();
+  }
+
+  @override
+  void didUpdateWidget(MenuAcceleratorLabel oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.label != oldWidget.label) {
+      _updateDisplayLabel();
+    }
+  }
+
+  static bool _altIsPressed() {
+    return HardwareKeyboard.instance.logicalKeysPressed.intersection(
+      <LogicalKeyboardKey>{
+        LogicalKeyboardKey.altLeft,
+        LogicalKeyboardKey.altRight,
+        LogicalKeyboardKey.alt,
+      },
+    ).isNotEmpty;
+  }
+
+  bool _handleKeyEvent(KeyEvent event) {
+    assert(_platformSupportsAccelerators());
+    final bool altIsPressed = _altIsPressed();
+    if (altIsPressed != _showAccelerators) {
+      setState(() {
+        _showAccelerators = altIsPressed;
+        _updateAcceleratorShortcut();
+      });
+    }
+    // Just listening, does't ever handle a key.
+    return false;
+  }
+
+  void _updateAcceleratorShortcut() {
+    assert(_platformSupportsAccelerators());
+    _shortcutRegistryEntry?.dispose();
+    _shortcutRegistryEntry = null;
+    // Before registering an accelerator as a shortcut it should meet these
+    // conditions:
+    //
+    // 1) Is showing accelerators (i.e. Alt key is down).
+    // 2) Has an accelerator marker in the label
+    // 3) Has an associated action callback for the label (from the
+    //    MenuAcceleratorCallbackBinding)
+    // 4) Is part of an anchor that either doesn't have a submenu, or doesn't
+    //    have any submenus currently open (only the "deepest" open menu should
+    //    have accelerator shortcuts registered).
```

Comment:
Nit: Inconsistent periods here on these numbered points.  I think they should all end with periods?